web/src/dialog_widget.ts

```diff
@@ -98,21 +98,10 @@ export function hide_dialog_spinner(): void {
 
 export function show_dialog_spinner(): void {
     const dialog_widget_selector = current_dialog_widget_selector();
-    // Disable both the buttons.
-    $(`${dialog_widget_selector} .modal__btn`).prop("disabled", true);
-
     const $spinner = $(`${dialog_widget_selector} .modal__spinner`);
-    const dialog_submit_button_span_width = $(".dialog_submit_button span").width();
-    const dialog_submit_button_span_height = $(".dialog_submit_button span").height();
-
-    // Hide the submit button after computing its height, since submit
-    // buttons with long text might affect the size of the button.
-    $(".dialog_submit_button span").hide();
 
-    loading.make_indicator($spinner, {
-        width: dialog_submit_button_span_width,
-        height: dialog_submit_button_span_height,
-    });
+    // Use the common spinner handling function
```

Comment:
This comment is useless, it's obvious from the name `loading.show_spinner` that this is doing what you describe in this comment.